### DISCUSSION TITLE
ELK 8 integrations

### DISCUSSION THREAD

**Discussion by hacknisty** [wazuh/wazuh#17641] [General]
Hi,

I would like to deploy Wazuh on my infrastructure and was reviewing the different way to setup the whole thing.

I already have a ELK 8.8 cluster collecting system logs. This cluster is also used to retrieve applications logs accros my infrastructure.

What would be your recommandations on setting up Wazuh ?

Should I remove my current system log shipping, and replace it with Wazuh ? 
Or Is there a way to use my current indexes with Wazuh ?

I would find it a bit overkill, to also use the wazuh indexer on a single node when I already have an ES cluster.

After searching a bit, i could see there is work done for ELK 8 Integrations but can't find anything stable or any proper documentation on the current state of ELK 8 integrations.

Regards,

**Comment by mdiego92** [comment]
Hello  @hacknisty,

There is no Wazuh plugin compatible with ELK 8.8 at the moment.

- Available plugins: https://github.com/wazuh/wazuh-kibana-app/wiki/Compatibility
- Documentation for reference: https://documentation.wazuh.com/current/upgrade-guide/compatibility-matrix/index.html

If you want to use the Wazuh plugin to manage your Wazuh deployment, you will need a Kibana whose version has a compatible Wazuh plugin (see the compatibility matrix).

- Building a custom Kibana plugin: https://documentation.wazuh.com/current/development/packaging/generate-wazuh-kibana-app.html

E.g.:
`./generate_wazuh_app.sh -b 4.4.4-8.8-dev -s ./wazuhapp/kibana/out -r 1`

There are two main issues with the compatibility:

- Adapt the building of the visualizations related to Elasticsearch data
The previous way to build the visualizations is not compatible with Kibana 8.8.x. We researched how to render these visualizations and did a POC with some of them where we saw that this could be the way to create them. There are some unsolved problems as the connection of the search bar with the visualizations and vice versa required to ensure the same functionality. More information: https://github.com/wazuh/wazuh-kibana-app/issues/4567#issuecomment-1307409214. Adapting the visualizations and dashboards could require a significant amount of time.

- Fix the problems in the Events tab of the modules
We identified some problems and saw the possible solutions for some of them, but there are others that are not solved. We can not check if there are more problems until we solve the previous problems.

It is important to know that after solving the mentioned problems, others could appear when exploring other sections or features of the Wazuh plugin. Unfortunately, some of the unsolved problems block us to check other plugin functionalities, for this reason, we need to resolve them before proceeding.

___

Another option that may be useful, depending on your case, is Syslog Output.
Documentation for reference:

- https://documentation.wazuh.com/current/user-manual/manager/manual-syslog-output.html
- https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/syslog-output.html#reference-ossec-syslog-output

Please let me know if you have any remaining questions!
Looking forward to your feedback!

**Reply by hacknisty** [reply]
Hello @mdiego92 ,

Thank you very much for all those informations.

Well my ELK 8.8 cluster was fairly new, and considering everything, I choose to downgrade our ELK 8
.8 to the last ELK 7 supported by Wazuh.

I followed official documentation on how to setup Wazuh with ELK cluster. Everything went fine and smooth and is working properly now.

Wazuh is an awesome tool, and i'm really glad to have it working. Thank you (and the Wazuh team of course) again for all your work !

Best regards,

**Reply by mdiego92** [reply]
Hello @hacknisty,

Great! Thank you very much for the feedback.
Please let us know if you have any issues or questions :smile:!!
